wgengine/magicsock/magicsock.go

```diff
@@ -1138,6 +1138,15 @@ func (c *Conn) sendUDP(ipp netip.AddrPort, b []byte) (sent bool, err error) {
 		if sent {
 			metricSendUDP.Add(1)
 		}
+
+		if stats := c.stats.Load(); stats != nil {
+			c.mu.Lock()
+			ep, ok := c.peerMap.endpointForIPPort(ipp)
+			c.mu.Unlock()
```

Comment:
extra mutex acquire here isn't ideal

Reply:
I think we could plumb this through from Send() it's got a copy of the conn.Endpoint interface.

Reply:
Will take a look at that!

Reply:
I've looked at the paths that leaves to this call, and they seem to be quite rare, so there might be a case for omitting recording it:

`sendUDP` is only called from `sendAddr`, but only if the address is not `tailcfg.DerpMagicIPAddr`.

`sendAddr` is called from `send` in endpoint only if it is a derp address, otherwise `sendUDPBatch` is called, meaning it will not reach `sendUDP`

`sendAddr` is called from `sendDiscoMessage` where the `dst` might be a UDP addr.

I assume connstats should also log the discos going over UDP, if we don't need to, then we can omit this code, if we do, there is quite a few calls to `sendDiscoMessage` that will need `ep` plumbed through.

Thoughts?